impeller/renderer/backend/vulkan/resource_manager_vk.h

```diff
@@ -120,19 +163,30 @@ class UniqueResourceVKT {
 
   ~UniqueResourceVKT() { Reset(); }
 
-  const ResourceType* operator->() const { return resource_.get()->Get(); }
+  /// @brief      Returns a pointer to the resource.
+  const ResourceType* operator->() const {
+    // If this would segfault, fail with a nicer error message.
+    FML_CHECK(resource_) << "UniqueResourceVKT was reclaimed.";
+
+    return resource_.get()->Get();
+  }
 
-  void Reset(ResourceType&& other) {
+  /// @brief      Reclaims the existing resource, if any, and replaces it.
+  ///
+  /// @param[in]  other   The (new) resource to move.
+  void Swap(ResourceType&& other) {
```

Comment:
I read through this a bit more after you'd mentioned it.  It is pretty weird using an rvalue here and storing it as a unique_ptr.  Seems like taking a unique_ptr would make more sense, but it's not a huge deal.

Reply:
I'm supportive of changing, that was some of the confusion I had the first time!

Reply:
Will defer for now, doesn't seem super important. Ack.